### PR TITLE
More concise Perl style; verbosity levels; use hashes and anonarrays …

### PR DESCRIPTION
**Pull request by 0-ast-0**
More legwork: these were thing I would have changed had I not wanted the pkgsrc diff to remain small.

We have better debugging with -vvv now.

I follow Perl Best Practices an dmake sure the code passes percritic. For that reason all the shift commands are now gone (for example).

### CODE REVIEW

File: fgallery [hunk 1]
```diff
@@ -138,82 +124,74 @@ sub is_fgallery_dir
 sub slurp
 {
   my ($fn) = @_;
-  open(my $fd, '<', $fn) or fatal("cannot read $fn: $!");
-  if ($^V lt v5.23.4) {
-    binmode($fd, ":encoding($ENCODING)");
-  }
+  no warnings; # avoid wide character warnings
```
Comment (kensanata) [inline fgallery:127]: Is there no way to get rid of the wide character warnings by using the encoding layer in the open call?

> open(my $fd, '<:raw', $fn)

or

> open(my $fd, "<:encoding($ENCODING)", $fn)

or simply default to utf8?

File: fgallery [hunk 2]
```diff
@@ -410,17 +385,17 @@ sub print_help
...
   -c methods            caption extraction methods (txt,xmp,exif,cmt, or none)
   -o                    do not auto-orient
   -k                    do not modify files, keep original
   -t                    do not time-sort
   -r                    reverse album order
-  -u                    update existing gallery
+  -u                    update existing gallery with additional files
   -p                    do not automatically include full-sized panoramas
-  -d                    do not generate a full album download
+  -d                    do not create a full album ZIPfile for download
```
Comment (kensanata) [inline fgallery:398]: How about just calling it a zip file? Isn't that common usage these days?

**Comment by kensanata** [comment]
I'm going to merge anyway because I like it and because I'm not quite sure what do about the wide character warnings.

**Comment by 0-ast-0** [comment]
On Tue, Mar 13, 2018 at 06:23:37AM +0000, Alex Schroeder wrote:
> kensanata commented on this pull request.
> >    -p                    do not automatically include full-sized panoramas
> -  -d                    do not generate a full album download
> +  -d                    do not create a full album ZIPfile for download
> 
> How about just calling it a zip file? Isn't that common usage these days?

Yes you're right, fixed.

**Pull request merged**